### PR TITLE
[WIP] Adds a way to create a olc::Pixel from a hsv color

### PR DESCRIPTION
**Pull request by Lett1**
Adds a new static function called `FromHsv` that allows you to specify a color in hsv space.

### CODE REVIEW

File: olcPGEX_AdditionalColours.h [hunk 1]
```diff
@@ -60,5 +60,60 @@ namespace olc
...
+		// Creates a new Pixel instance from hsv space (hue, saturation and value).
+		// Hue ranges from 0 to 360, saturation and value from 0 to 1.
+		static olc::Pixel FromHsv(float hue, float saturation, float value, const float alpha = 1)
+		{
+			hue = std::clamp(hue, 0.0f, 360.0f);
+			saturation = std::clamp(saturation, 0.0f, 1.0f);
+			value = std::clamp(value, 0.0f, 1.0f);
+			
+			const float chroma = value * saturation;
+			const float x = chroma * (1 - std::fabs(fmodf((hue / 60), 2) - 1));
```
Comment (matt-hayward) [inline olcPGEX_AdditionalColours.h:74]: What are `x` and `m`? Probably worth giving them more descriptive names.

File: olcPGEX_AdditionalColours.h [hunk 2]
```diff
@@ -60,5 +60,60 @@ namespace olc
...
+			const float m = value - chroma;
+
+			float rawRed, rawGreen, rawBlue;
+			
+			if (hue < 60)
```
Comment (matt-hayward) [inline olcPGEX_AdditionalColours.h]: It's probably a bit of pedantry, but could you put the opening and closing braces for the if-else on the same line, please?

E.g:

```cpp
if (hue < 60) {
    // etc
} else if {
    // etc
} else {
    // etc
}

**Review by matt-hayward — CHANGES_REQUESTED** [review]

File: olcPGEX_AdditionalColours.h (hunk 1 above)
Comment (Lett1) [reply olcPGEX_AdditionalColours.h:74]: x appears to be the second largest component of the color, m is used to match the raw colors to the given value. Most implementations and writeups don't give them better names either.

File: olcPGEX_AdditionalColours.h (hunk 2 above)
Comment (Lett1) [reply olcPGEX_AdditionalColours.h]: Done

**Pull request merged**